packages/playwright-utils/src/screenshots/types.ts

```diff
@@ -0,0 +1,84 @@
+import type AxeBuilder from "@axe-core/playwright";
+import type { expect, MountResultJsx, test } from "@playwright/experimental-ct-vue";
+import type { JSX } from "vue/jsx-runtime";
+
+export type UseMatrixScreenshotTestOptions = {
+  /**
+   * Test function to use. Must support a axe fixture for accessibility testing.
+   *
+   * @see https://playwright.dev/docs/accessibility-testing#using-a-test-fixture-for-common-axe-configuration
+   */
+  test: ReturnType<typeof test.extend<AxeFixture>>;
+  /**
+   * Playwright expect function to use.
+   *
+   * @example `import { expect } from "@playwright/experimental-ct-vue";`
+   */
+  expect: typeof expect;
+  /**
+   * Global default options for the matrix screenshot tests.
+   * Will be merged with the options passed to a single screenshot test.
+   */
+  defaults?: Partial<
+    Pick<MatrixScreenshotTestOptions, "disabledAccessibilityRules" | "disablePadding">
+  >;
+};
+
+export type AxeFixture = {
+  makeAxeBuilder: () => AxeBuilder;
+};
+
+export type MatrixScreenshotTestOptions<
+  TColumn extends string = string,
+  TRow extends string = string,
+> = {
+  /**
+   * Test name. Will be displayed above the matrix screenshot and be used as filename.
+   */
+  name: string;
+  /**
+   * Matrix columns.
+   */
+  columns: readonly TColumn[];
+  /**
+   * Matrix rows.
+   */
+  rows: readonly TRow[];
+  /**
+   * Function that returns the component for the given column and row.
+   */
+  component: (column: TColumn, row: TRow) => JSX.Element;
+  /**
+   * Optional callback to be executed before capturing the screenshot.
+   * Useful for performing `expect()` or e.g. hover, focus-visible state etc.
+   * Focus and mouse will be reset after each screenshot.
+   */
+  beforeScreenshot?: ScreenshotTestHook<TColumn, TRow>;
```

Comment:
Would prefer to rename or clarify in the description that this is called before an **individual/single** screenshot is taken
```suggestion
  beforeScreenshot?: ScreenshotTestHook<TColumn, TRow>;
```